projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -423,20 +417,36 @@ export class DatagridComponent<R> implements OnInit, AfterViewInit, OnDestroy {
     /**
      * Returns the items selected in the VCD datagrid.
      */
-    get datagridSelection(): R[] {
+    @Input()
+    get datagridSelection(): (B | R)[] {
```

Comment:
`B | R` sounds wrong if `R` extends `B`. Shouldn't it just be `B`? And we can specify that callers must cast and test if if they want to use `R` since we can't guarantee it... 

If this is really required, we should define a type instead of copy pasting `(R | B)[]`